elasticapm/context/contextvars.py

```diff
@@ -50,16 +56,60 @@ def get_transaction(self, clear=False):
             return None
 
     def set_transaction(self, transaction):
+        """
+        Set the transaction for the current execution context
+        """
         self.elasticapm_transaction_var.set(transaction)
 
-    def get_span(self):
-        try:
-            return self.elasticapm_span_var.get()
-        except LookupError:
-            return None
+    def get_span(self, extra=False):
+        """
+        Get the active span for the current execution context.
+
+        If extra=True, a tuple will be returned with the span and its extra
+        data: (span, extra)
+        """
+        spans = self.elasticapm_span_var.get()
+        span = (None, None)
+        if spans:
+            span = spans[-1]
+        if extra:
+            return span
+        else:
+            return span[0]
+
+    def set_span(self, span, extra=None):
+        """
+        Set the active span for the current execution context.
+
+        The previously-activated span will be saved to be re-activated later.
+
+        Optionally, `extra` data can be provided and will be saved alongside
+        the span.
+        """
+        self.elasticapm_spans_var.set(self.elasticapm_span_var.get() + (span, extra))
```

Comment:
Shouldn't this be

```suggestion
        self.elasticapm_spans_var.set(self.elasticapm_span_var.get() + ((span, extra),))
```
Adding two tuples creates a new tuple that is the concatenation of the two tuples.

Reply:
You're totally right, good catch.